### ISSUE TITLE
Not connecting

### ISSUE THREAD

**Issue by Arthurhoeke** [SteamRE/SteamKit#320]
Hi,

I've been trying to use SteamKit2, and I found some sample code on github for logging in.
Now, the problem is is that after I receive the message that says it is connecting to steam nothing happends. I don't get any new messages, nothing happens.
Code:

http://pastebin.com/3H1Si1Wi

**Comment by Netshroud** [comment]
Can you enable `DebugLog` and post it's output? See Sample 3 for an example.

**Comment by VoiDeD** [comment]
The most obvious issue I see is that you're running your callback loop in a UI thread event handler. This will effectively block the UI thread, which might be the issue you're encountering.

**Comment by Arthurhoeke** [comment]
After posting the issue I had, I decided to downgrade the steamkit version I was using to 1.5 I believe it was. I updated my code to the older version, and now it seems to be working sometimes. When I connect, it says that it is connecting to steam and sometimes it progresses and sometimes it just seems to be stuck. I guess that's better than what I had before. And what VoiDeD said, I do have the problem that my UI freezes. I guess I'll have to make a call for the callback in a new void.
Also, kinda off-topic I want to change a label in my UI, but because I need to change it inside of static code I can't do it, not sure how to fix that too.
